src/utils.coffee

```diff
@@ -18,4 +18,17 @@ class Utils
     return Math.random() unless seed
     # a MUCH simplified version inspired by PlanOut.js
     parseInt(@sha1(seed).substr(0, 13), 16) / 0xFFFFFFFFFFFFF
+  @checkWeights: (variants) ->
+    contains_weight = []
+    contains_weight.push(value.weight?) for key, value of variants
+    contains_weight.every (has_weight) -> has_weight
+  @sumWeights: (variants) ->
+    sum = 0
+    for key, value of variants
+      sum += value.weight if value.weight?
+    sum
+  @validateWeights: (variants) ->
+    contains_weight = []
```

Comment:
again, it's a matter of personal preference, but I would find it clearer to write something like

```coffee
variants_with_weights = variants.filter((variant) -> variant.weight).length
return true if variants_with_weights == 0 || variants.length == variants_with_weights
false
```

(then you can also re-use `variants_with_weights` as a shared function